internal/storage/utils.go

```diff
@@ -837,3 +839,56 @@ func GetPkFromInsertData(collSchema *schemapb.CollectionSchema, data *InsertData
 
 	return realPfData.Data, nil
 }
+
+func boolFieldDataToPbBytes(field *BoolFieldData) ([]byte, error) {
+	arr := &schemapb.BoolArray{Data: field.Data}
+	return proto.Marshal(arr)
+}
+
+func stringFieldDataToPbBytes(field *StringFieldData) ([]byte, error) {
+	arr := &schemapb.StringArray{Data: field.Data}
+	return proto.Marshal(arr)
+}
+
+func binaryWrite(endian binary.ByteOrder, data interface{}) ([]byte, error) {
+	buf := new(bytes.Buffer)
+	err := binary.Write(buf, endian, data)
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+// FieldDataToBytes encode field data to byte slice.
+// For some fixed-length data, such as int32, int64, float vector, use binary.Write directly.
+// For binary vector, return it directly.
+// For bool data, first transfer to schemapb.BoolArray and then marshal it. (TODO: handle bool like other scalar data.)
+// For variable-length data, such as string, first transfer to schemapb.StringArray and then marshal it.
+// TODO: find a proper way to store variable-length data. Or we should unify to use protobuf?
+func FieldDataToBytes(endian binary.ByteOrder, fieldData FieldData) ([]byte, error) {
+	switch field := fieldData.(type) {
+	case *BoolFieldData:
+		// return binaryWrite(endian, field.Data)
+		return boolFieldDataToPbBytes(field)
+	case *StringFieldData:
+		return stringFieldDataToPbBytes(field)
+	case *BinaryVectorFieldData:
+		return field.Data, nil
+	case *FloatVectorFieldData:
+		return binaryWrite(endian, field.Data)
+	case *Int8FieldData:
+		return binaryWrite(endian, field.Data)
+	case *Int16FieldData:
+		return binaryWrite(endian, field.Data)
+	case *Int32FieldData:
+		return binaryWrite(endian, field.Data)
+	case *Int64FieldData:
+		return binaryWrite(endian, field.Data)
+	case *FloatFieldData:
+		return binaryWrite(endian, field.Data)
+	case *DoubleFieldData:
+		return binaryWrite(endian, field.Data)
```

Comment:
Nit: maybe use short form like:
```
case *FloatVectorFieldData, *Int8FieldData, *Int16FieldData...
      return binaryWrite(endian, field.Data)
```

Reply:
did not notice it was type cast switch, please ignore this comment

Reply:
The types of field is not same here.